rainbows.rb

```diff
@@ -27,3 +31,28 @@ def mix_colors(color1,color2)
 
 new_color = mix_colors(red,green)
 puts "#{color_lookup[red]} #{red} + #{color_lookup[green]} #{green} = #{color_lookup[new_color]} #{new_color}"
+# Status API Training Shop Blog About
+
+rainbow = [{"red" => {r: 255, g: 0, b: 0}}, {"orange" => {r: 255, g: 125, b: 0}}, {"yellow" => {r: 255, g: 255,b: 0}},
+{"green" => {r: 0, g: 255, b:0}}, {"blue" => {r: 0, g: 0, b: 255}}, {"indigo" => {r: 75, g: 0, b: 130}},
+{"violet" => {r: 143, g: 0, b: 255}}]
+
+rainbow.each do |i|
+  i.each do |color, code|
+    if color == "red"
+      puts "#{color}".to_s.color(:red)
+    elsif color == "orange"
+      puts "#{color}".to_s.color(:orange)
+    elsif color == "yellow"
+      puts "#{color}".to_s.color(:yellow)
+    elsif color == "green"
+      puts "#{color}".to_s.color(:green)
+    elsif color == "blue"
+      puts "#{color}".to_s.color(:blue)
+    elsif color == "indigo"
+      puts "#{color}".to_s.color(:indigo)
+    else color == "violet"
+      puts "#{color}".to_s.color(:violet)
+    end
+  end
+end
```

Comment:
This is good, but does hard code in the colorings instead of generalizing as I showed in class.